cou/utils/openstack.py

```diff
@@ -147,3 +230,98 @@ def lookup(cls, component: str, version: str) -> List[str]:
             if version in version_range:
                 compatible_os_releases.append(openstack_release)
         return compatible_os_releases
+
+
+class OpenStackRelease:
+    """Provides a class that will compare strings from an iterator type object.
+
+    Used to provide > and < comparisons on strings that may not necessarily be
+    alphanumerically ordered.  e.g. OpenStack or Ubuntu releases AFTER the
+    z-wrap.
+    """
+
+    openstack_codenames = list(OPENSTACK_CODENAMES.keys())
+    openstack_release_year = list(OPENSTACK_CODENAMES.values())
+
+    def __init__(self, os_release: str):
+        """Initialize the OpenStackRelease object.
+
+        :param os_release: OpenStack release codename.
+        :type os_release: str
+        :raises ValueError: Raises ValueError if OpenStack codename is unknown.
+        """
+        self.os_release = os_release
+        try:
+            self.index = self.openstack_codenames.index(os_release)
+        except ValueError as exc:
+            raise ValueError(
+                f"OpenStack '{os_release}' is not in '{self.openstack_codenames}'"
+            ) from exc
```

Comment:
not-blocking: I think it would be nicer to os_release.setter, so the init will look cleaner + I can do `OpenStackRelease.os_release = "my new os release"`.